utils/project.go

```diff
@@ -0,0 +1,129 @@
+	language, buildType := "unknown", "docker"
+	if _, err := os.Stat(path.Join(projectPath, "pom.xml")); err == nil {
+		language = "java"
+		buildType = determineJavaBuildType(projectPath)
+	}
+	if _, err := os.Stat(path.Join(projectPath, "package.json")); err == nil {
+		language = "nodejs"
+		buildType = "nodejs"
+	}
+	if _, err := os.Stat(path.Join(projectPath, "Package.swift")); err == nil {
+		language = "swift"
+		buildType = "swift"
+	}
+	return language, buildType
+}
+
+// CheckProjectPath will stop the process and return an error if path does not
+// exist or is invalid
+func CheckProjectPath(projectPath string) {
+	if projectPath == "" {
+		log.Fatal("Project path has not been set")
+	}
+
+	if _, err := os.Stat(projectPath); os.IsNotExist(err) {
+		log.Fatal("Project not found at given path")
+	}
+}
+
+func determineJavaBuildType(projectPath string) string {
+	pathToPomXML := path.Join(projectPath, "pom.xml")
+	pomXMLContents, _err := ioutil.ReadFile(pathToPomXML)
+	// if there is an error reading the pom.xml, we build as docker
+	if _err != nil {
+		return "docker"
+	}
+	pomXMLString := string(pomXMLContents)
+	if strings.Contains(pomXMLString, "<groupId>org.springframework.boot</groupId>") {
+		return "spring"
+	}
+	if strings.Contains(pomXMLString, "<groupId>org.eclipse.microprofile</groupId>") {
+		return "liberty"
+	}
+	return "docker"
+}
+
+// WriteNewCwSettings writes a default .cw-settings file to the given path,
+// dependant on the build type of the project
+func WriteNewCwSettings(pathToCwSettings string, BuildType string,) {
+	defaultCwSettings := getDefaultCwSettings()
+	cwSettings := addNonDefaultFieldsToCwSettings(defaultCwSettings, BuildType)
+	settings, err := json.MarshalIndent(cwSettings, "", "")
+	errors.CheckErr(err, 203, "")
+	err = ioutil.WriteFile(pathToCwSettings, settings, 0644)
+}
+
+// RenameLegacySettings renames a .mc-settings file to .cw-settings
+func RenameLegacySettings(pathToLegacySettings string, pathToCwSettings string) {
+	err := os.Rename(pathToLegacySettings, pathToCwSettings)
+	errors.CheckErr(err, 205, "")
+}
+
+func getDefaultCwSettings() CWSettings {
+	return CWSettings{
+		ContextRoot:  "",
+		InternalPort: "",
+		HealthCheck:  "",
+		IgnoredPaths: []string{""},
+	}
+}
+
+func addNonDefaultFieldsToCwSettings(cwSettings CWSettings, ProjectType string) CWSettings {
+	projectTypesWithInternalDebugPort := []string{"liberty", "spring", "nodejs"}
+	projectTypesWithMavenSettings := []string{"liberty", "spring"}
+	if stringInSlice(ProjectType, projectTypesWithInternalDebugPort) {
+		// We use a pointer, as an empty string would be removed due to omitempty on struct
+		defaultValue := ""
+		cwSettings.InternalDebugPort = &defaultValue
+	}
+	if stringInSlice(ProjectType, projectTypesWithMavenSettings) {
+		cwSettings.MavenProfiles = []string{""}
+		cwSettings.MavenProperties = []string{""}
+	}
+	return cwSettings
+}
+
+func stringInSlice(a string, list []string) bool {
```

Comment:
nit: given this is called `stringInSlice`, maybe `list` should be renamed to `slice`?